oracle/src/main/java/org/jdbi/v3/oracle/OracleReturning.java

```diff
@@ -56,83 +58,65 @@
  * }
  * </pre>
  * Though you can bind multiple params, and whatnot
+ * <p>
+ * This class is beta, and may be changed incompatibly or removed at any time.
  */
-@Deprecated
-public class OracleReturning<ResultType> implements StatementCustomizer
-{
+public class OracleReturning<ResultType> implements StatementCustomizer {
     private final RowMapper<ResultType> mapper;
     private final List<int[]> binds = new ArrayList<>();
     private StatementContext context;
     private List<ResultType> results;
-    private Class<?> oraclePS ;
-    private Method registerReturnParameter ;
-    private Method getReturnResultSet ;
-    private Object stmt ;
+    private OraclePreparedStatement stmt;
 
     /**
      * Provide a mapper which knows how to do positional access, sadly the
      * {@link BeanMapper} uses the names in the result set
      *
      * @param mapper Must use only positional access to the result set
      */
-    public OracleReturning(RowMapper<ResultType> mapper)
-    {
+    public OracleReturning(RowMapper<ResultType> mapper) {
         this.mapper = mapper;
-        try {
-            this.oraclePS = Class.forName("oracle.jdbc.OraclePreparedStatement");
-            this.registerReturnParameter = oraclePS.getMethod("registerReturnParameter", new Class[]{int.class, int.class});
-            this.getReturnResultSet = oraclePS.getMethod("getReturnResultSet");
-        }
-        catch (Exception e) {
-            throw new RuntimeException(e);
-        }
     }
 
     /**
      * @see StatementCustomizer#beforeExecution(java.sql.PreparedStatement, StatementContext)
      */
     @Override
-    public void beforeExecution(PreparedStatement stmt, StatementContext ctx) throws SQLException
-    {
+    public void beforeExecution(PreparedStatement stmt, StatementContext ctx) throws SQLException {
         this.context = ctx;
 
-        if (!oraclePS.isAssignableFrom(stmt.getClass())) {
+        if (stmt instanceof OraclePreparedStatement) {
+            this.stmt = OraclePreparedStatement.class.cast(stmt);
+        } else {
             try {
-                final Method get_delegate = stmt.getClass().getMethod("getDelegate");
-                final Object candidate = get_delegate.invoke(stmt);
-                if (!oraclePS.isAssignableFrom(candidate.getClass())) {
+                // If it's an instance of Apache commons-dbcp DelegatingPreparedStatement
+                final Method getDelegate = stmt.getClass().getMethod("getDelegate");
```

Comment:
`Statement` itself extends `Wrapper`, and dbcp appears to implement it right, so.. yes.